code/game/objects/items/stacks/packagewrap.dm

```diff
@@ -0,0 +1,207 @@
+/obj/item/stack/package_wrap
+	name = "package wrap"
+	desc = "Wrapping paper designed to help goods safely navigate the mail system."
+	icon = 'icons/obj/items.dmi'
+	icon_state = "deliveryPaper"
+	singular_name = "paper sheet"
+	w_class = 2
+	amount = 24
+	max_amount = 24
+	//If it's null, it can't wrap that type.
+	var/smallpath = /obj/item/smallDelivery //We use this for items
+	var/bigpath = /obj/structure/bigDelivery //We use this for structures (crates, closets, recharge packs, etc.)
+	var/manpath = null //We use this for people.
+	var/human_wrap_speed = 100 //Handcuffs are 30
+
+	var/list/cannot_wrap = list(
+		/obj/structure/table,
+		/obj/structure/rack,
+		/obj/item/smallDelivery,
+		/obj/structure/bigDelivery,
+		/obj/item/weapon/gift,
+		/obj/item/weapon/winter_gift,
+		/obj/item/weapon/evidencebag,
+		/obj/item/weapon/legcuffs/bolas,
+		/obj/item/weapon/storage
+		)
+
+	var/list/wrappable_big_stuff = list(
+		/obj/structure/closet,
+		/obj/structure/vendomatpack,
+		/obj/structure/stackopacks
+		)
+
+/obj/item/stack/package_wrap/afterattack(var/attacked, mob/user as mob)

```

Comment:
Should be `/obj/item/stack/package_wrap/afterattack(var/atom/attacked, var/mob/user, var/proximity_flag)`

Then check if `proximity_flag` is true, because right now this has no adjacency checks.